src/Access/EnabledQuota.cpp

```diff
@@ -66,7 +64,12 @@ struct EnabledQuota::Impl
             UInt64 n = static_cast<UInt64>((current_time - end + duration) / duration);
             end = end + duration * n;
             if (end_of_interval.compare_exchange_strong(end_loaded, end.time_since_epoch()))
+            {
+                /// We reset counters only if the interval's end has been calculated before.
+                /// If it hasn't we just calculate the interval's end for the first time and don't reset counters yet.
+                need_reset_counters = true;
```

Comment:
```suggestion
                need_reset_counters = (end_loaded.count() != 0);
```

I mean we shouldn't reset counters if `end_loaded.count() == 0` here (see the failed tests). There can be three cases at the point `getEndOfInterval()` is called:
1) `current_time < end_of_interval` -- the counters should not be reset.
2) `current_time >= end_of_interval` -- the counters usually should be reset, with an exception if we have the case 3 (see below).
3) `end_of_interval == 0` -- that means the `end_of_interval` is not initialized. Of course in this case also `current_time >= end_of_interval` because `current_time >= 0` for any `current_time`. But in this case that doesn't mean we should reset the counters: `end_of_interval` is actually not initialized, so we can't say that the `current_time >= end_of_interval` does really means anything in this case.

Reply:
ok，thank you for advice